src/components/TicketModal/TicketModal.vue

```diff
@@ -0,0 +1,272 @@
+<template>
+  <div class="ticket-modal">
+    <div
+      class="ticket-modal-header"
+      :style="{ 'background-color': computedStatusColor }"
+    >
+      <div class="logo">
+        <IconSluggo :height="50" :width="50" />
+      </div>
+      <div
+        class="ticket-name"
+        :style="{ 'background-color': computedStatusColor }"
+      >
+        <div class="editableText">
+          <div v-if="!isEditing">
+            <span class="text" @click="enableEditing">
+              {{ ticketTitle }}
+            </span>
+          </div>
+          <div v-if="isEditing" @focusout="disableEditing">
+            <input
+              v-model="tempText"
+              class="input text-box"
+              v-bind:style="{ 'background-color': computedStatusColor }"
+              v-focus
+            />
+          </div>
+        </div>
+      </div>
+      <div class="editable-icon" v-if="shouldShowPencil">
+        <i class="bx bx-pencil"></i>
+      </div>
+      <div class="dropdown ticket-status" :class="statusDropdownClass">
+        <div class="dropdown-trigger" @click="toggleStatusDropdown">
+          <button
+            class="button is-primary"
+            :style="{ 'background-color': computedStatusColor }"
+            aria-haspopup="true"
+            aria-controls="dropdown-menu"
+          >
+            <span>{{ ticketStatus }}</span>
+            <span class="icon is-small">
+              <i class="bx bx-chevron-down"></i>
+            </span>
+          </button>
+        </div>
+        <div class="dropdown-menu">
+          <div class="dropdown-content">
+            <div
+              class="dropdown-item"
+              v-if="ticketStatus != 'To Do'"
+              @click="setTicketStatus('To Do')"
```

Comment:
I don't think it's wise to do this sort of `v-if` chain as statuses are customizable by team. Ideally, you should be making a list of possible statuses from the return of the api call and select based on those values. For now, change this so that it renders from a list of statuses.

Reply:
Ah I didn't realize this was an option for the user